src/cargo/ops/cargo_test.rs

```diff
@@ -167,10 +168,18 @@ fn run_doc_tests(
 
         config.shell().status("Doc-tests", unit.target.name())?;
         let mut p = compilation.rustdoc_process(unit, *script_meta)?;
-        p.arg("--test")
-            .arg(unit.target.src_path().path().unwrap())
-            .arg("--crate-name")
-            .arg(&unit.target.crate_name());
+        p.arg("--crate-name").arg(&unit.target.crate_name());
+        p.arg("--test");
+
+        if nightly_features_allowed() {
```

Comment:
☝️ @jyn514 This is what I mean; removing this, every test that does a doctest will fail when run against stable rustc (up until it rides the trains)

Reply:
I don't understand the problem; why do you want to remove `if nightly_features_allowed`? That seems like the right check.

If it helps, cargo always runs against the same toolchain version of rustc; if cargo is nightly, rustc will be nightly too.

Reply:
I mean, this should get into stable at some point…

Reply:
Oh - that has to wait for rustdoc to stabilize the option. The plan is

- [x] add the nightly feature in rustdoc
- [ ] pass it automatically in nightly cargo (this PR)
- [ ] stabilize the feature in rustdoc
- [ ] pass the flag unconditionally in cargo

Reply:
I think that this should probably be gated by a specific `-Z` flag in Cargo rather than simply unconditionally on the nightly channel

Reply:
That'll also make the impact in the tests less because the new changes will have to be opted-into instead of automatically happening on nightly.

Reply:
:+1: fair enough! Rebased and added an unstable option that controls this behavior. That should be well enough for my purposes.